algoworld_contracts/swapper/swap_proxy.py

```diff
@@ -0,0 +1,119 @@
+"""
+MIT License
+
+Copyright (c) 2022 AlgoWorld
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import dataclasses
+import sys
+
+from pyteal import (
+    Addr,
+    And,
+    Balance,
+    Bytes,
+    Cond,
+    Global,
+    Gtxn,
+    Int,
+    Mode,
+    Or,
+    Substring,
+    Txn,
+    TxnType,
+    compileTeal,
+)
+
+from algoworld_contracts.common.utils import parse_params
+
+"""
+Swapper Proxy Used for Storing Swap Configurations
+1. Activate Proxy
+2. Store Swap Configurations
+"""
+
+TEAL_VERSION = 6
+
+STORE_GSIZE = Int(2)
+STORE_FEE = 0
+STORE_PROXY_NOTE = 1
+
+
+@dataclasses.dataclass
+class SwapProxy:
+    swap_creator: str
+    version: str
+
+
+def swapper_proxy(cfg: SwapProxy):
+
+    is_proxy_store = And(
+        Global.group_size() == STORE_GSIZE,
+        Gtxn[STORE_FEE].type_enum() == TxnType.Payment,
+        Gtxn[STORE_PROXY_NOTE].type_enum() == TxnType.Payment,
+    )
+
+    return Cond([is_proxy_store, proxy_store(cfg)])
+
+
+def proxy_store(cfg: SwapProxy):
+    Balance(Txn.sender())
+
+    store_fee = And(
+        Gtxn[STORE_FEE].type_enum() == TxnType.Payment,
+        Or(
+            Gtxn[STORE_FEE].amount() == Int(110_000),
+            Gtxn[STORE_FEE].amount() == Int(10_000),
```

Comment:
Doubt: why should a user pay `110_000` if paying just `10_000` is approved by the contract?

Reply:
@cusma pooled fees sounds cool actually wasnt aware of it. I did wanted to have it atomic though to minimize the amount of different transaction groups/transactions that id force user to sign. In this case this 110_000 really only happens once for any wallet doing first ever swapper, afterwards UI just checks for the state. But i agree this can be decoupled and i could actually tie it with the wallet during first logging and user would have a prompt popup asking to sign proxy config transaction to be able to use swapper (that's also where i can put terms & conditions checkbox). 

I am a bit tied with the release plan though. As usual I will move your comment to github issue and will get back to it once time allows. Thanks for the suggestion as usual!

Reply:
Resolving the issue (as the request for change is out of scope for initial release)